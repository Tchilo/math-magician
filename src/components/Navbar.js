import React from 'react';
import { Link } from 'react-router-dom';

const Navbar = () => {
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Home',
    },
    {
      id: 2,
      path: '/App',
      text: 'Calulator',
    },
    {
      id: 3,
      path: '/Quote',
      text: 'Quote',
    },
  ];

  return (
    <nav data-testId="navbar" className="navBar">
      <h2>Math Magicians</h2>
      <ul className="links">
        {links.map((link) => <li key={link.id}><Link to={link.path} activeClassName="active-link" exact>{link.text}</Link></li>)}
      </ul>
    </nav>
  );
};

export default Navbar;
