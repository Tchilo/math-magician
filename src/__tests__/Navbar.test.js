import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Navbar from '../components/Navbar';

test('renders as expected', () => {
  const tree = renderer.create(<Navbar />).toJSON();
  expect(tree).toMatchSnapshot();
});
