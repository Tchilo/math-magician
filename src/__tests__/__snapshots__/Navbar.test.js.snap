// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<nav
  className="navBar"
  data-testid="navbar"
>
  <h2>
    Math Magicians
  </h2>
  <ul
    className="links"
  >
    <li>
      <a
        activeClassName="active-link"
        exact={true}
        href="#/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        activeClassName="active-link"
        exact={true}
        href="#/App"
        onClick={[Function]}
      >
        Calulator
      </a>
    </li>
    <li>
      <a
        activeClassName="active-link"
        exact={true}
        href="#/Quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
